File: Algorithms/cpp/SquaresOfASortedArray/squaresOfASortedArray.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<int> sortedSquares(std::vector<int>& nums) {
        // Map each element 'num' in vector 'nums' to its square and store the result in a new vector
        // Sort the new vector in non-decreasing order
        // Return the sorted vector
        std::vector<int> squared;
        for (int num : nums) {
            squared.push_back(num * num);
        }
        std::sort(squared.begin(), squared.end());
        return squared;
    }
};

File: Algorithms/python/squaresOfASortedArray.py
def sortedSquares(nums):

    return sorted([a ** 2 for a in nums])
class Solution(object):
    def sortedSquares(self, nums):
        # Map each element 'a' in array 'A' to its square and store the result in a new array
        # Sort the new array in non-decreasing order
        # Return the sorted array
        return sorted([a ** 2 for a in nums])
